packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -76,15 +85,47 @@ export interface EditableTree {
     readonly [proxyTargetSymbol]: object;
 
     /**
-     * Anchor to this node.
-     * Valid as long as this EditableTree's context is not freed.
-     * Might not point to any node if this node is deleted from the document.
-     *
-     * TODO: When a proper editing API is exposed on EditableTree directly,
-     * this should become an implementation detail and rbe removed from this API surface.
+     * Creates a child of this node.
+     */
+    readonly [insertNodeSymbol]: (key: string, value: ITreeCursor) => boolean;
+
+    /**
+     * Sets value of a child of this node.
+     */
+    readonly [setValueSymbol]: (key: string, value: unknown, typeName: TreeSchemaIdentifier) => boolean;
+
+    /**
+     * Deletes a child of this node.
      */
-    readonly [anchorSymbol]: Anchor;
+    readonly [deleteNodeSymbol]: (key: string) => boolean;
+}
 
+/**
+ * A tree where the forest has no data neither an initialized root node.
+ * By calling a function associated with `insertRootSymbol` is converted into EditableTree.
+ */
+export interface EmptyEditableTree {
+    readonly [insertRootSymbol]: (root: ITreeCursor) => EditableTree | UnwrappedEditableSequence;
+    /**
+     * Stores the target for the proxy which implements reading and writing for this node.
+     * The details of this object are implementation details,
+     * but the presence of this symbol can be used to separate EditableTrees from other types.
+     */
+    readonly [proxyTargetSymbol]: object;
+}
+
+/**
+ * A tree which can be traversed and edited.
+ *
+ * When iterating, only visits non-empty fields.
+ * To discover empty fields, inspect the schema using {@link getTypeSymbol}.
+ *
+ * TODO: `extends Iterable<EditableField>`
+ * TODO: provide non-schema impacted APIs for getting fields and nodes without unwrapping
+ * (useful for generic code, and when references to these actual fields and nodes are required,
+ * for example creating anchors and editing).
+ */
+export interface EditableTree extends FieldlessEditableTree {
```

Comment:
EditableTrees can have fields, so having all EditableTrees also be FieldlessEditableTrees sounds wrong/confusing.

I think FieldlessEditableTree should be renamed. To describe the subset of a tree node's API that only talks about the data on that node and not its children, we use the type "NodeData". A similar name would make sense here, maybe EditibleNodeData.